apps/aecontract/src/aect_dispatch.erl

```diff
@@ -58,78 +58,57 @@ encode_call_data(_, _, _, _) ->
+run_common(#{  amount      := Value
+             , beneficiary := <<Beneficiary:?BENEFICIARY_PUB_BYTES/unit:8>> = BeneficiaryBin
+             , call        := Call
+             , call_data   := CallData
+             , call_stack  := CallStack
+             , caller      := <<CallerAddr:?PUB_SIZE/unit:8>>
+             , code        := Code
+             , contract    := <<Address:?PUB_SIZE/unit:8>> = ContractPubKey
+             , difficulty  := Difficulty
+             , gas         := Gas
+             , gas_price   := GasPrice
+             , height      := Height
+             , time        := Time
+             , trees       := Trees
+             }, VMVersion) ->
+    ConsensusVersion = aec_hard_forks:protocol_effective_at_height(Height),
+    TxEnv = aetx_env:contract_env(Height, ConsensusVersion, Time,
+                                  BeneficiaryBin, Difficulty),
+    ChainState0 = aec_vm_chain:new_state(Trees, TxEnv, ContractPubKey),
+    Env = #{currentCoinbase   => Beneficiary,
+            currentDifficulty => Difficulty,
+            %% TODO: implement gas limit in governance and blocks.
+            currentGasLimit   => 100000000000,
+            currentNumber     => Height,
+            currentTimestamp  => Time,
+            chainState        => ChainState0,
+            chainAPI          => aec_vm_chain,
+            vm_version        => VMVersion},
+    Exec = #{code       => Code,
+             address    => Address,
+             caller     => CallerAddr,
+             data       => CallData,
+             gas        => Gas,
+             gasPrice   => GasPrice,
+             origin     => CallerAddr,
+             value      => Value,
+             call_stack => CallStack
+            },
     Spec = #{ env => Env,
-              exec => #{},
+              exec => Exec,
               pre => #{}},
-    call_common(CallDef, Spec).
-
-call_AEVM_01_Solidity_01(#{ contract    := ContractPubKey
-                          , height      := Height
-                          , trees       := Trees
-                          , beneficiary := <<Beneficiary:?BENEFICIARY_PUB_BYTES/unit:8>>
-                          } = CallDef) ->
-    Env = set_env(ContractPubKey, Height, Trees, Beneficiary, aec_vm_chain, ?AEVM_01_Solidity_01),
-    Spec = #{ env => Env,
-              exec => #{},
-              pre => #{}},
-    call_common(CallDef, Spec).
-
-set_env(ContractPubKey, Height, Trees, Beneficiary, API, VmVersion) ->
-    ChainState = aec_vm_chain:new_state(Trees, Height, ContractPubKey),
-    #{currentCoinbase   => Beneficiary,
-      %% TODO: get the right difficulty (Tracked as #159522571)
-      currentDifficulty => 0,
-      %% TODO: implement gas limit in governance and blocks.
-      currentGasLimit   => 100000000000,
-      currentNumber     => Height,
-      %% TODO: should be set before starting block candidate.
-      currentTimestamp  => aeu_time:now_in_msecs(),
```

Comment:
I see this was ["Set timestamp from current block timestamp in contract state."](https://www.pivotaltracker.com/n/projects/2159793/stories/159522630) - addressed by this change. Fed from `aec_chain_state` (and `aec_block_micro_candidate`).